samza-azure/src/main/java/org/apache/samza/system/eventhub/producer/EventHubSystemProducer.java

```diff
@@ -109,27 +108,23 @@ public EventHubSystemProducer(EventHubConfig config, String systemName,
     this.registry = registry;
     this.systemName = systemName;
     this.partitioningMethod = config.getPartitioningMethod(systemName);
-    this.eventHubClientManagerFactory = eventHubClientManagerFactory;
     this.interceptors = interceptors;
+
+    List<String> streamNames = config.getStreams(systemName);
+    // Create and initiate connections to Event Hubs
```

Comment:
1. We should be clear on what the semantic meaning of `streamId` and `physicalName` is. In Samza, a `streamId` is the identifier for the stream. It is used to identify a stream (including stream-specific properties) in config. 
2. We are key-ing this map of eventHubClients by their streamId, but looking them up by their physicalName later when send is called. Should be consistent with the identifiers we use.
Take a look at `StreamConfig` to obtain a physicalName for a given streamId.
3. Also, see if we can get rid off this property - "systems.eventhubs.streams". The set of streams should be derivable from other ways by using `StreamConfig.getStreamIds()`. Having said that, I'm fine if we have to keep it for compatibility reasons for now.